features/support/env.rb

```diff
@@ -1,54 +1,82 @@
-require 'nokogiri'
+require 'browsermob/proxy'
+require 'capybara/chromedriver/logger'
 require 'capybara/cucumber'
-require 'capybara/poltergeist'
-require 'uri'
+require 'nokogiri'
 require 'plek'
+require 'selenium-webdriver'
+require 'uri'
 
-ENV["GOVUK_WEBSITE_ROOT"] ||= "https://www-origin.integration.publishing.service.gov.uk"
-ENV["GOVUK_DRAFT_WEBSITE_ROOT"] ||= Plek.new.external_url_for('draft-origin')
-
-non_production_website_roots = [
-  "https://www-origin.integration.publishing.service.gov.uk",
-  "https://www-origin.staging.publishing.service.gov.uk",
-  "https://www-origin.integration.govuk.digital",
-  "https://www-origin.blue.integration.govuk.digital",
-  "https://www-origin.green.integration.govuk.digital",
-  "https://www-origin.staging.govuk.digital",
-  "https://www-origin.blue.staging.govuk.digital",
-  "https://www-origin.green.staging.govuk.digital",
-]
+if ENV["AUTH_USERNAME"] && ENV["AUTH_PASSWORD"]
+  basic_auth_credentials = "#{ENV['AUTH_USERNAME']}:#{ENV['AUTH_PASSWORD']}@"
+end
 
-case ENV["GOVUK_WEBSITE_ROOT"]
-when *non_production_website_roots
-  ENV["EXPECTED_GOVUK_WEBSITE_ROOT"] = ENV["GOVUK_WEBSITE_ROOT"]
+# Set up environment
+case ENV["ENVIRONMENT"]
+when "integration"
+  ENV["GOVUK_APP_DOMAIN"] ||= "integration.publishing.service.gov.uk"
+  ENV["GOVUK_WEBSITE_ROOT"] ||= "https://www-origin.integration.publishing.service.gov.uk"
+  ENV["GOVUK_WEBSITE_ROOT_WITH_AUTH"] ||= "https://#{basic_auth_credentials}www-origin.integration.publishing.service.gov.uk"
+when "staging"
+  ENV["GOVUK_APP_DOMAIN"] ||= "staging.publishing.service.gov.uk"
+  ENV["GOVUK_WEBSITE_ROOT"] ||= "https://www-origin.staging.publishing.service.gov.uk"
+when "production"
+  ENV["GOVUK_APP_DOMAIN"] ||= "publishing.service.gov.uk"
+  ENV["GOVUK_WEBSITE_ROOT"] ||= "https://www.gov.uk"
 else
-  ENV["EXPECTED_GOVUK_WEBSITE_ROOT"] = 'https://www.gov.uk'
+  raise "ENVIRONMENT should be one of integration, staging or production"
 end
 
-Capybara.app_host = ENV["GOVUK_WEBSITE_ROOT"]
-phantomjs_logger = File.open("log/phantomjs.log", "a")
+# Set up basic URLs
+ENV["GOVUK_DRAFT_WEBSITE_ROOT"] ||= Plek.new.external_url_for("draft-origin")
+Capybara.app_host = ENV["GOVUK_WEBSITE_ROOT_WITH_AUTH"] || ENV["GOVUK_WEBSITE_ROOT"]
 
-GOOGLE_ANALYTICS_URL = 'www.google-analytics.com'
-BLACKLISTED_URLS = [
-  GOOGLE_ANALYTICS_URL,
-  'www.youtube.com'
-]
+# Set up proxy server (used to manipulate HTTP headers etc since Selenium doesn't
```

Comment:
I've added a section to the README to explain why we have the proxy. I need to check whether it will run on the monitoring machine but I think Java is installed everywhere so we should be OK.